src/Idris/Elab/Interface.idr

```diff
@@ -238,6 +232,20 @@ getMethToplevel {vars} env vis iname cname constraints allmeths params sig
     methName : Name -> Name
     methName n = UN (Basic $ bindName n)
 
+    collectImplicitNames : RawImp -> List Name
+    collectImplicitNames (IPi _ _ Explicit _        _ ty) = []
+    collectImplicitNames (IPi _ _ _        (Just n) _ ty) = n :: collectImplicitNames ty
+    collectImplicitNames (IPi _ _ _        Nothing  _ ty) = collectImplicitNames ty
+    collectImplicitNames _                                = []
+
+    namesToRawImp : Bool -> List Name -> RawImp -> RawImp
```

Comment:
This function is based on [a similar function from `processRecord`](https://github.com/idris-lang/Idris2/blob/8e420d6b2e29c276f77b562955f01958664ccb58/src/TTImp/ProcessRecord.idr#L300-L302). It may be worth making a global function for it